src/sparseml/pytorch/utils/trainer.py

```diff
@@ -0,0 +1,562 @@
+import os
+from typing import Optional, Union, Dict, Any, Callable, Tuple, List
+import warnings
+
+from sparseml.pytorch.datasets import DatasetRegistry, ssd_collate_fn, \
+    yolo_collate_fn
+from sparseml.pytorch.image_classification.utils import save_recipe, \
+    save_model_training
+from sparseml.pytorch.models import ModelRegistry
+from sparseml.pytorch.optim import ScheduledModifierManager, ScheduledOptimizer
+from sparseml.pytorch.sparsification import ConstantPruningModifier
+from sparseml.pytorch.utils import (
+    torch_distributed_zero_first,
+    set_deterministic_seeds,
+    LossWrapper,
+    SSDLossWrapper,
+    YoloLossWrapper,
+    TopKAccuracy,
+    CrossEntropyLossWrapper,
+    InceptionCrossEntropyLossWrapper,
+    default_device,
+    TensorBoardLogger,
+    PythonLogger,
+    model_to_device,
+    ModuleTrainer,
+    ModuleDeviceContext,
+    ModuleTester,
+    DEFAULT_LOSS_KEY,
+    get_prunable_layers,
+    tensor_sparsity,
+)
+from sparseml.utils import create_dirs
+from sparsezoo import Zoo
+import torch
+import logging
+
+from torch import Module
+from torch.utils.data import DataLoader, Dataset
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ImageClassificationTrainer:
+    "Utility class for training image classification models"
+
+    def __init__(
+            self,
+            checkpoint_path: str,
```

Comment:
we definitely need to cut down on the number of arguments we have here. for example, the dataloaders should be able to be constructed outside of the trainer and then passed in. Additionally, let's also keep model saving outside of the trainer for now. and can you remind me on the blocks we saw for model construction needing to be in the trainer class?